compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -326,34 +344,36 @@ class TreePickler(pickler: TastyPickler) {
   def pickleDef(tag: Int, mdef: MemberDef, tpt: Tree, rhs: Tree = EmptyTree, pickleParams: => Unit = ())(using Context): Unit = {
     val sym = mdef.symbol
 
-    assert(symRefs(sym) == NoAddr, sym)
-    registerDef(sym)
-    writeByte(tag)
-    val addr = currentAddr
-    try
-      withLength {
-        pickleName(sym.name)
-        pickleParams
-        tpt match {
-          case _: Template | _: Hole => pickleTree(tpt)
-          case _ if tpt.isType => pickleTpt(tpt)
+    if assertForBestEffort(symRefs.get(sym) == Some(NoAddr) && !(tag == TYPEDEF && tpt.isInstanceOf[Template] && !tpt.symbol.exists)) then
```

Comment:
This particular incantation probably deserves a comment and/or a local `def` to give a name to what we're testing here.